api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1980,4 +1980,43 @@ public boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType)
 		}
 		return mappedClasses;
 	}
+
+	@Override
+	public Concept getConceptByUuidOrMapping(String uuid, String code, String sourceName) throws APIException {
+		return getConceptByUuidOrMapping(uuid, code, sourceName, true);
+	}
+
+	@Override
+	public Concept getConceptByUuidOrMapping(String uuid, String code, String sourceName, boolean includeRetired)
+	        throws APIException {
+		List<Concept> concepts = Context.getConceptService().getConceptsByUuidOrMapping(uuid,code, sourceName, includeRetired);
+		
+		if (concepts.isEmpty()) {
+			return null;
+		}
+		else if (concepts.size() > 1 && !concepts.get(1).getRetired()) {
+			throw new APIException("Concept.error.multiple.non.retired", new Object[] { code, sourceName });
+		} else {
+			return concepts.get(0);
+					}	
+	}
+
+	@Override
+	public List<Concept> getConceptsByUuidOrMapping(String uuid, String code, String sourceName) throws APIException {
+		return getConceptsByUuidOrMapping(uuid, code, sourceName, true);
+	}
+
+	@Override
+	public List<Concept> getConceptsByUuidOrMapping(String uuid, String code, String sourceName, boolean includeRetired)
+	        throws APIException {
+		if(uuid==null && code==null) {
+			throw new IllegalArgumentException("Specify at least uuid or code with its sourceName or all");	
+		}else if(uuid!=null ) {
+			getConceptByUuid(uuid);				
+			}
+		else if(code!=null && sourceName!=null){
+			getConceptByMapping(code, sourceName);
+		}	
+		return null;
```

Comment:
@reagan-meant This method will always return null. You are not returning anything from the getConceptByUuid and getConceptByMapping method calls

Reply:
check in the api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java
these two methods are already there and they have return values....

Reply:
@reagan-meant You not returning the result from the method calls. for example in line 2018 you are calling the method getConceptByMapping but do not return the result from it.

Reply:
@dilanthas thank you, I have noticed it...